migrate.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+from typing import List, Optional, Tuple, Dict, Any
+
+import os
+import argparse
+import subprocess
+import datetime
+import json
+
+ZFS_SNAPSHOTDIR = '.zfs/snapshot'
+
+SNAPSHOT_TAG = "snapshot="
+
+DEBUG = False
+
+
+def _run(command: str, input: Optional[str] = None, void_stderr: bool = False) -> None:
+    other_args = dict()
+    if void_stderr and not DEBUG:
+        other_args["stderr"] = subprocess.DEVNULL
+    subprocess.run(command, shell=True, text=True, input=input, **other_args)
+
+
+def _eval(command: str, input: Optional[str] = None, void_stderr: bool = False) -> str:
```

Comment:
You should take `command` as a list of arguments and then run without shell. This way you don't have to escape anything. And we don't need a shell around the process anyway.